recipes/pcre2/all/conanfile.py

```diff
@@ -29,6 +29,7 @@ class PCRE2Conan(ConanFile):
         "with_bzip2": [True, False],
         "support_jit": [True, False],
         "grep_support_callout_fork": [True, False],
+        "link_size": [2, 3, 4],
```

Comment:
Sorry for the little experience in prce2, have to ask: is 4 the max just because having more than 4GB of resulting compiled expression would be overkill, or is it a hard limit in the codebase?

Reply:
The number ultimately represents the number of bytes to use to represent offsets in the compiled expression. Which sounds like branch sizes in assembly but I think is more for the state machine opcodes to be able to say "how far to jump in the event of some matching state in order to find the next proper state." So if that number is only 2 bytes (16 bits), the next state can't be more than 65535 bytes away. And if it's a 2MB expression, that's going to often be way too far to represent. The trade off though is that there's more "fluff" in the state machine opcodes so more cache thrashing => worse performance where those larger jumps weren't needed.

That said, why it can't go up to 8 for example, I don't know. Having looked through the code, it seems like it maybe could. But I might be missing some place where a value ultimately is read into a uint32_t or something. In either case, the autconf based build strictly enforces that it's 2, 3, or 4. I don't think the cmake one does though.

Reply:
Thanks for the insight! :)